.gitignore

```diff
@@ -5,53 +5,73 @@
 # created during manual debugging and many people like to clean up instead of
 # having git ignore such leftovers. You can use `.git/info/exclude` to
 # configure your local ignore list.
-# FIXME: This needs cleanup.
-*~
+
+# File system
+.DS_Store
+desktop.ini
+
+# Version control
+.hgignore
+.hg/
```

Comment:
Does it make sense to include these? We use git, not mercurial.

Reply:
They were part of the [very first git commit](https://github.com/rust-lang/rust/commit/c01efc669f09508b55eced32d3c88702578a7c3e), when switching the project over from mercurial. Now 11 years later I assume indeed no-one is using mercurial for this project anymore :), so I can remove them.

Reply:
Looking at that commit, `/Makefile` looks like another candidate to be removed (since we now use `x.py`). However there I'm much less certain that it is not used in someones workflow today.

Reply:
@CDirkx Makefile should still be ignored, it's generated by `./configure`.